liquibase-couchbase/src/test/java/liquibase/ext/couchbase/changelog/CouchbaseHistoryServiceTest.java

```diff
@@ -0,0 +1,436 @@
+
+        List<RanChangeSet> returnedChangeLogs = couchbaseHistoryService.getRanChangeSets();
+
+        assertEquals(expectedRanChangeSets, returnedChangeLogs);
+    }
+
+    @Test
+    @SneakyThrows
+    void Should_return_empty_ran_changesets_when_collection_not_exists() {
+        CouchbaseConnection connection = mock(CouchbaseConnection.class);
+        Cluster cluster = mock(Cluster.class);
+        BucketManager bucketManager = mock(BucketManager.class);
+        Bucket bucket = mock(Bucket.class);
+        CollectionManager collectionManager = mock(CollectionManager.class);
+        ScopeSpec scopeSpec = mock(ScopeSpec.class);
+        List<ScopeSpec> scopeSpecs = singletonList(scopeSpec);
+        CollectionSpec collectionSpec = mock(CollectionSpec.class);
+        Set<CollectionSpec> collectionSpecs = new HashSet<>();
+        collectionSpecs.add(collectionSpec);
+
+        when(database.getConnection()).thenReturn(connection);
+        when(connection.getCluster()).thenReturn(cluster);
+        when(cluster.buckets()).thenReturn(bucketManager);
+        when(bucketManager.getBucket(SERVICE_BUCKET_NAME)).thenReturn(mock(BucketSettings.class));
+        when(cluster.bucket(SERVICE_BUCKET_NAME)).thenReturn(bucket);
+
+        when(bucket.collections()).thenReturn(collectionManager);
+        when(collectionManager.getAllScopes()).thenReturn(scopeSpecs);
+        when(scopeSpec.collections()).thenReturn(collectionSpecs);
+        when(collectionSpec.scopeName()).thenReturn(DEFAULT_SERVICE_SCOPE);
+        when(collectionSpec.name()).thenReturn("another collection");
+
+        List<RanChangeSet> returnedChangeLogs = couchbaseHistoryService.getRanChangeSets();
+
+        assertTrue(returnedChangeLogs.isEmpty());
+    }
+
+    @Test
+    @SneakyThrows
+    void Should_return_the_same_ran_changesets_if_already_returned() {
+        ChangeSet changeSet = prepareChangeSet();
+        RanChangeSet ranChangeSet = new RanChangeSet(changeSet, ChangeSet.ExecType.EXECUTED, null, null);
+        List<RanChangeSet> expectedRanChangeSets = singletonList(ranChangeSet);
+
+        couchbaseHistoryService.setRanChangeSetList(expectedRanChangeSets);
+
+        List<RanChangeSet> returnedChangeLogs = couchbaseHistoryService.getRanChangeSets();
+
+        assertEquals(expectedRanChangeSets, returnedChangeLogs);
+        verify(changeLogOperator, never()).getAllChangeLogs();
+    }
+
+    @Test
+    @SneakyThrows
+    void Should_return_ran_changeset_when_collection_exists() {
+        ChangeSet changeSet = prepareChangeSet();
+        RanChangeSet expectedRanChangeSet = new RanChangeSet(changeSet, ChangeSet.ExecType.EXECUTED, null, null);
+        List<RanChangeSet> expectedRanChangeSets = singletonList(expectedRanChangeSet);
+        couchbaseHistoryService.setRanChangeSetList(expectedRanChangeSets);
+
+        CouchbaseConnection connection = mock(CouchbaseConnection.class);
```

Comment:
Is it possible to move some of test configuration on class instance level? I find that there are too many configurations in some test methods that duplicates each other

Reply:
Yes. I wanted to move to seperate method, but forgot) thank you

Reply:
Done